octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/DescriptionComposite.java

```diff
@@ -1,69 +1,160 @@
+import com.hpe.octane.ideplugins.eclipse.ui.util.resource.PlatformResourcesManager;
+import com.hpe.octane.ideplugins.eclipse.util.EntityFieldsConstants;
+
+public class DescriptionComposite extends Composite {
+
+    private PropagateScrollBrowserFactory factory = new PropagateScrollBrowserFactory();
+    private Color foregroundColor = PlatformResourcesManager.getPlatformForegroundColor();
+    private Color backgroundColor = PlatformResourcesManager.getPlatformBackgroundColor();
+    private Browser browserDescHtml;
+    private File octanePhotosDir;
+
+    public DescriptionComposite(Composite parent, int style) {
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
+        StackLayoutComposite stackLayoutComposite = new StackLayoutComposite(this, SWT.NONE);
+        browserDescHtml = factory.createBrowser(stackLayoutComposite, SWT.NONE);
+        browserDescHtml.addLocationListener(new LinkInterceptListener());
+        stackLayoutComposite.showControl(browserDescHtml);
+    }
+
+    public void setEntityModel(EntityModelWrapper entityModelWrapper) {
+        browserDescHtml.setText(getBrowserText(entityModelWrapper.getReadOnlyEntityModel()));
+    }
+
+    public String getBrowserText(EntityModel entityModel) {
+        String descriptionFromServerRemodeled = Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION)));
+        try {
+            descriptionFromServerRemodeled = downloadPictures(
+                    Util.getUiDataFromModel(entityModel.getValue((EntityFieldsConstants.FIELD_DESCRIPTION))));
+        } catch (IOException e) {
+        }
+
+        String descriptionText = "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">"
+                + descriptionFromServerRemodeled
+                + "</font></body></html>";
+        if (descriptionText.equals("<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                + getRgbString(foregroundColor) + ">" + "</font></body></html>")) {
+
+            return "<html><body bgcolor =" + getRgbString(backgroundColor) + ">" + "<font color ="
+                    + getRgbString(foregroundColor) + ">" + "No description" + "</font></body></html>";
+        } else {
+            return descriptionText;
+        }
+
+    }
+
+    private static String getRgbString(Color color) {
+        return "rgb(" + color.getRed() + "," + color.getGreen() + "," + color.getBlue() + ")";
+    }
+
+    private String downloadPictures(String descriptionField) throws IOException {
+        String tmpPath = System.getProperty("java.io.tmpdir");
+        String baseUrl = Activator.getConnectionSettings().getBaseUrl();
+        File tmpDir = new File(tmpPath);
+        octanePhotosDir = new File(tmpDir, "Octane_pictures");
+
+        if (!new File(tmpDir, "Octane_pictures").exists()) {
```

Comment:
@vladbenchea Is "new File(tmpDir, "Octane_pictures")" necessary here since we just declared "octanePhotosDir" just above?

Reply:
I did it this way to force-try a new File and recheck if it exists. Not sure if it was the best solution, but it seemed like it when I wrote the code.